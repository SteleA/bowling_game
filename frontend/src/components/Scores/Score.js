import React from 'react';
import _ from 'lodash';

const score = (props) => {
  const isExtra = props.rolls.length === 3;
  return (
    <div className={`score ${props.active ? 'active' : ''}`}>
      <span className="round">{props.round}</span>
      <div className={`score-body ${isExtra ? 'extra-roll' : ''}`}>
        {_.map(props.rolls, roll =>
          <span
            key={_.uniqueId()}
            className="score-box"
          >{roll.score}</span>)}
        <div className="currentScore">{props.score}</div>
      </div>
    </div>
  );
};

export default score;
