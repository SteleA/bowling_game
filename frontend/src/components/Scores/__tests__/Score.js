import { shallow } from 'enzyme';  // eslint-disable-line
import { shallowToJson } from 'enzyme-to-json';  // eslint-disable-line
import React from 'react';
import Score from '../Score';

const props = {
  rolls: [],
};

describe('<Score />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Score {...props} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
