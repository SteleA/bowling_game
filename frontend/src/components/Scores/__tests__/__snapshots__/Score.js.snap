// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Score /> should render with default props 1`] = `
<div
  className="score "
>
  <span
    className="round"
  />
  <div
    className="score-body "
  >
    <div
      className="currentScore"
    />
  </div>
</div>
`;
